src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -117,6 +202,36 @@ public Uri GetEndpointForClient(ConfigurationClient client)
             ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
             
             return currentClient?.Endpoint;
+        }
+
+
+        // Only remove the client if it is not in the user passed connection string or endpoints, as well as not in returned SRV records.
+        private bool IsEligibleToRemove(IEnumerable<string> srvEndpointHosts, ConfigurationClientWrapper client)
+        {
+            if (_connectionStrings != null && _connectionStrings.Any(c => GetHostFromConnectionString(c).Equals(client.Endpoint.Host, StringComparison.OrdinalIgnoreCase)))

```

Comment:
This would parse the endpoint from each connection string everytime we invoke `IsEligibleToRemove`.
How about storing the list of endpoints instead of connection strings? Looks like we only use the first connection string anyway. Maybe we can store that one connection string separately and extract endpoint from all other connection strings before storing in `_endpoint` list.
That also makes it simpler to check `isUsingConnectionString`. We can avoid the check for `_connectionStrings.Any()` and just check if the single connection string is set.

Reply:
I have updated this, could you please tale a look again?